scenarios/kubernetes_e2e.py

```diff
@@ -485,18 +493,21 @@ def main(args):
 
     if args.use_shared_build is not None:
         # find shared build location from GCS
-        build_file = ''
-        if args.use_shared_build:
-            build_file += args.use_shared_build + '-'
-        build_file += 'build-location.txt'
-        gcs_path = os.path.join(args.gcs_shared, os.getenv('PULL_REFS', ''), build_file)
+        gcs_path = get_gcs_path(args)
         print >>sys.stderr, 'Getting shared build location from: '+gcs_path
-        try:
-            # tell kubetest to extract from this location
-            args.kubetest_args.append('--extract=' + read_gcs_path(gcs_path))
-            args.build = None
-        except urllib2.URLError as err:
-            raise RuntimeError('Failed to get shared build location: %s' % err.reason)
+        retries = 0
+        while True:
```

Comment:
maybe for/else will be more clear here

Reply:
hmm ok `for _ in xrange(10):` could work, but I'm not sure if that's clearer.

Reply:
I think the main problem is it makes using the actual error ugly. I want to ignore the error the first n times and then give up and give the error.